model/src/main/java/org/activityinfo/model/query/BitSetColumnView.java

```diff
@@ -98,8 +100,22 @@ public ColumnView select(int[] selectedRows) {
 
     @Override
     public int[] order(int[] sortVector, SortModel.Dir direction, int[] range) {
-        // TODO: BitSet Sorting
-        // Do not sort on column
+        switch(direction) {
+            case ASC:
+                if (range == null || range.length == numRows) {
+                    HeapsortColumn.heapsortBitSet(bitSet, sortVector, numRows, true);
+                } else {
+                    HeapsortColumn.heapsortBitSet(bitSet, sortVector, range.length, range, true);
```

Comment:
Can we collapse these two branches and just pass the parameter (direction == Direction.ASC) ?

Reply:
Yes I actually did that with the latest commit, but I should update the rest as well.